src/editor/block/commands.js

```diff
@@ -41,7 +61,45 @@ export const addNewBlockAfterCurrent = ({ state, dispatch }) => {
         },
         selection: EditorSelection.cursor(block.content.to + delimText.length)
     }, {
-        scrollIntoView: true, 
+        scrollIntoView: true,
+        userEvent: "input",
+    }))
+    return true;
+}
+
+export const addNewBlockBeforeFirst = ({ state, dispatch }) => {
+    if (state.readOnly)
+        return false
+    const block = getFirstNoteBlock(state)
+    const delimText = "\n∞∞∞text-a\n"
+
+    dispatch(state.update({
+        changes: {
+            from: block.delimiter.from,
+            insert: delimText,
+        },
+        selection: EditorSelection.cursor(delimText.length) // TODO: fix jump. See first char of file when saving
```

Comment:
There was a few times where your solution didn't work. But feel free to run the tests with your solution, and try manually as well. FWIR it was a matter of having two `"\n∞∞∞text-a\n"` lines and messing up the beginning of the file.

There might also be a visual bug that I can't catch with Playwright, when `addNewBlockBeforeFirst` is called when the top of the file is already on screen, hence the comment. I'll try to post a screenshot.

Reply:
So I have been able to reproduce the visual bug that I had the other day.

How to reproduce:
- Scroll at the top of the buffer
- Select a block that is visible form the screen
- Trigger the creation of a new block before the first one

When I am further down the buffer, so that the first block isn't visible, the block doesn't happen.

![image](https://github.com/heyman/heynote/assets/4939807/d42a0d5f-1d0b-411a-84bf-c5d3a13278a8)

The first line appears smaller.

Scrolling down then up fixes it.

Reply:
@fabjan Marybe I'm doing something wrong, but could you test on your side with your version, what happens when trying to create a block at the top when the first block is manually set as `Markdown` or `typescript` ?

For me it seems that if the syntax name is longer than a (yet undefined) threshold, it breaks...

Reply:
I've pushed a fix (9ff77428edd26d442042337856061cbd3c682b8e) for the visual bug.